### PR TITLE
Update to use NUnit test adapter

### PR DESCRIPTION
**Pull request by slang25**

### CODE REVIEW

File: src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj [hunk 1]
```diff
@@ -19,21 +18,24 @@
...
-  </ItemGroup> 
+  </ItemGroup>
   <ItemGroup>
     <PackageReference Include="FakeItEasy" Version="3.4.2" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
     <PackageReference Include="NUnit" Version="3.7.1" />
     <PackageReference Include="NUnitLite" Version="3.7.1" />
-  </ItemGroup> 
+    <PackageReference Include="NUnit3TestAdapter" Version="3.8.0" />
+    <PackageReference Include="Mono.Cecil" Version="0.10.0-beta6" />
```
Comment (AnthonySteele) [inline src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj]: Why do we need `Mono.Cecil` ?

Comment (slang25) [reply src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj]: @AnthonySteele Good question, I probably need a comment in there, it fails for me when I don't have this, it's a known issue with the new NUnit test adapter, this is a workaround. I'll dig out a link

Comment (slang25) [reply src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj]: https://github.com/nunit/nunit3-vs-adapter/issues/325

**Comment by slang25** [comment]
I'm confused by why it's not happy with the `NLog.config`, it's broken locally for me too.
I'll take a look at lunch.

**Comment by martincostello** [comment]
Weird how it's totally fine on Travis.

**Comment by slang25** [comment]
Yeah, it's proving tricky to debug as it works fine from VS test runner and R#, just breaks on `dotnet-test` and only on Windows. wtf?

Edit, I'm being silly, it was discovering the `net46` tests, if I remove `net46` R# can't see them and vs test breaks too.

**Comment by martincostello** [comment]
Just need to fix the path for the `.trx` upload in AppVeyor.

**Comment by slang25** [comment]
Turns out the `Debbuger` target is not supported for NetStandard 1.3, how it worked on travis is the greatest mystery known to man.

**Comment by martincostello** [comment]
👍

**Comment by slang25** [comment]
🎉

File: src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj [hunk 2]
```diff
@@ -19,21 +18,25 @@
...
-  </ItemGroup> 
+  </ItemGroup>
   <ItemGroup>
     <PackageReference Include="FakeItEasy" Version="3.4.2" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
     <PackageReference Include="NUnit" Version="3.7.1" />
     <PackageReference Include="NUnitLite" Version="3.7.1" />
-  </ItemGroup> 
+    <PackageReference Include="NUnit3TestAdapter" Version="3.8.0" />
+    <!-- Workaround for https://github.com/nunit/nunit3-vs-adapter/issues/325 -->
```
Comment (slang25) [inline src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj:28]: https://github.com/nunit/nunit3-vs-adapter/issues/366

**Comment by slang25** [comment]
I'm happy with this now, feedback welcome.

**Comment by martincostello** [comment]
Looks good to go to me.

File: src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj (hunk 2 above)
Comment (rprouse) [reply src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj:28]: You should also remove the NUnitLite reference...

**Pull request merged**